brandingbrand/flagship
# fix: Add 'Marquis' feature to Button with title dynamically reflecting CTA state

### Description
Reply on Issue/Reaquest https://github.com/brandingbrand/flagship/issues/340;

Now the button have two additional not required properties: 
`dynamicTitleStates: string[];`
This property require not empty string array that contains all title states you need;


`selectedTitleState: number;`
This property is responsible for the selected dynamicTitle index;

If the button title has no above mentioned props or props are invalid Button component should return the default title property;

### Test
1. Run test project on ios/android;
2. Click on any Item and open PDP page;
3. Click on Add to Cart button;
4. Observe;

_Expected result:_ 
Button should change the state to Added to Cart and then change state back to Add to Cart;
